source/code/plugins/out_oms_changetracking_file.rb

```diff
@@ -0,0 +1,591 @@
+module Fluent
+
+  class OutChangeTrackingFile < BufferedOutput
+
+    Plugin.register_output('out_oms_oms.changetracking.file', self)
+	
+    # Endpoint URL ex. localhost.local/api/
+
+    def initialize
+      super
+	  
+      require 'base64'
+      require 'digest'
+      require 'json'
+      require 'net/http'
+      require 'net/https'
+      require 'openssl'
+      require 'rexml/document'
+      require 'securerandom'
+      require 'socket'
+      require 'uri'
+      require_relative 'omslog'
+      require_relative 'oms_configuration'
+      require_relative 'oms_common'
+    end
+
+    config_param :omsadmin_conf_path, :string, :default => '/etc/opt/microsoft/omsagent/conf/omsadmin.conf'
+    config_param :cert_path, :string, :default => '/etc/opt/microsoft/omsagent/certs/oms.crt'
+    config_param :key_path, :string, :default => '/etc/opt/microsoft/omsagent/certs/oms.key'
+    config_param :proxy_conf_path, :string, :default => '/etc/opt/microsoft/omsagent/proxy.conf'
```

Comment:
Code copying is what I what I would like to avoid. I think what would be best is to create a filter plugin to perform any additional processing you want on the data, and then send it to the out_oms_blob output plugin so that you don't have to copy its methods.

Besides code quality, output plugin duplication should be avoided because each output plugin needs to be altered in configuration to be multi-homing compatible. In omsagent.conf, you can see that string replacement is currently used for the `<match oms.blob.**>` block, which is performed at onboarding time. If your plugin is not present at onboarding time (i.e. if it's delivered by DSC), then this gets trickier.